File: 剑指offer/剑指 Offer 63. 股票的最大利润.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int x =INT_MAX;
        int res=0;
        for(int i;i<prices.size();i++ ){
          //if(prices[i]<x) x=prices[i];
          x=min(x,prices[i]);
          res=  max(res,prices[i]-x);
        }
        return res;
    }
};

File: 剑指offer_java/剑指 Offer 63. 股票的最大利润.java
class Solution {
    public int maxProfit(int[] prices) {
        int tmp=Integer.MAX_VALUE, res=0;
        for(int price:prices){
            res=Math.max(res,price-tmp);
            tmp=Math.min(tmp,price);
        }
        return res;
    }
}
